### ISSUE TITLE
Reference for decomposition of coarser torch ops into finer torch ops.

### ISSUE THREAD

**Issue by pashu123** [llvm/torch-mlir#542]
https://github.com/pytorch/functorch/blob/main/functorch/_src/decompositions.py

This is a good reference for breaking/decomposing torch ops.

**Comment by pashu123** [comment]
Modified the softmax decomposition here: https://github.com/llvm/torch-mlir/pull/554

**Comment by pashu123** [comment]
Modified log softmax decomposition here: https://github.com/llvm/torch-mlir/pull/580

**Comment by silvasean** [comment]
Closing this as it is more "documentation" rather than an actual issue. (also the decompositions are now spread out between _decomps and _refs)
